azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/AbstractAzureMojo.java

```diff
@@ -339,6 +340,7 @@ protected Azure getOrCreateAzureClient() throws AzureAuthFailureException, Azure
         try {
             final MavenAuthConfiguration mavenAuthConfiguration = auth == null ? new MavenAuthConfiguration() : auth;
             mavenAuthConfiguration.setType(getAuthType());
+            SystemPropertyUtils.injectCommandLineParameter("auth", mavenAuthConfiguration, MavenAuthConfiguration.class);
```

Comment:
will `mavenAuthConfiguration` be null?
what happens if `mavenAuthConfiguration` is null, it seems your implementation does not support null.
and if yes, the third parameter is not needed.

Reply:
mavenAuthConfiguration should be always non-null, but the other parameter runtime may be null